src/NHibernate/SqlCommand/ANSIJoinFragment.cs

```diff
@@ -39,6 +40,12 @@ public override void AddJoin(string tableName, string alias, string[] fkColumns,
 
 			_fromFragment.Add(joinString + tableName + ' ' + alias + " on ");
 
+			if (fkColumns.Length == 0)
+			{
+				_fromFragment.Add(on);
+				return;
```

Comment:
Here, an additional adjustment should be done. `on` is usually generated with a leading `" and "`, which should be removed before adding to `_fromFragment`.